python-mip/mip_helper/mip_helper/io_helper.py

```diff
@@ -53,24 +58,50 @@ def fetch_data():
 
 def save_results(pfa, error, shape):
     """
-    Store algorithm results in the output DB.
+    Store algorithm results in the output DB. Update results if it already exists.
     :param pfa: PFA formatted results
     :param error: Error message (if any)
     :param shape: Result shape. For example: pfa_json.
     """
     engine = sqlalchemy.create_engine(_get_output_db_url())
 
-    sql = sqlalchemy.text("INSERT INTO job_result VALUES(:job_id, :node, :date, :pfa, :error, :shape, :function)")
+    # TODO: only temporary solution, final one will not be updating existing results
+    query = """
+    INSERT INTO job_result VALUES(:job_id, :node, :timestamp, :data, :error, :shape, :function)
+    ON CONFLICT (job_id, node) DO UPDATE
+      SET timestamp = :timestamp,
+          data = :data,
+          error = :error,
+          shape = :shape,
+          function = :function;
+    """
+    sql = sqlalchemy.text(query)
     engine.execute(sql,
                    job_id=_get_job_id(),
                    node=_get_node(),
-                   date=datetime.datetime.now(),
-                   pfa=pfa,
+                   # TODO: shouldn't this rather be utcnow()?
```

Comment:
probably, please try to use utcnow()

Reply:
I have added stats in Marigold-get-results-method branch

Reply:
Should I continue with that branch or rebase on top of it? (Can you push directly to my pull request or do I have to give you permissions for that?)